src/dnsperf.c

```diff
@@ -493,7 +504,22 @@ setup(int argc, char** argv, config_t* config)
         config->mode = perf_net_parsemode(mode);
 
     if (!server_port) {
-        server_port = config->mode == sock_dot ? DEFAULT_SERVER_DOT_PORT : DEFAULT_SERVER_PORT;
+        switch (config->mode) {
+        case sock_doh: 
+            server_port = DEFAULT_SERVER_DOH_PORT;
+            break;
+        case sock_dot: 
+            server_port = DEFAULT_SERVER_DOT_PORT;
+            break;
+        default:
+            server_port = DEFAULT_SERVER_PORT;
+            break;
+        }
+    }
+
+    if (config->mode == sock_doh) {
+        memcpy(net_doh_uri, doh_uri, strlen(doh_uri));
```

Comment:
This can be buffer overflowed.

Why are you using arrays? And why copy to another variable instead of setting it directly to `net_doh_uri`?

Reply:
fixed